pkg/engine/generation.go

```diff
@@ -1,22 +1,23 @@
 package engine
 
 import (
+	"time"
+
 	kyverno "github.com/kyverno/kyverno/pkg/api/kyverno/v1"
 	"github.com/kyverno/kyverno/pkg/engine/response"
 	"github.com/kyverno/kyverno/pkg/engine/variables"
 	"sigs.k8s.io/controller-runtime/pkg/log"
-	"time"
 )
 
 // Generate checks for validity of generate rule on the resource
 // 1. validate variables to be substitute in the general ruleInfo (match,exclude,condition)
 //    - the caller has to check the ruleResponse to determine whether the path exist
 // 2. returns the list of rules that are applicable on this policy and resource, if 1 succeed
-func Generate(policyContext PolicyContext) (resp *response.EngineResponse) {
-	return filterRules(policyContext)
+func Generate(policyContext PolicyContext, namespaceLabels map[string]string) (resp *response.EngineResponse) {
```

Comment:
Can we add `namespaceLabels ` to `PolicyContext `?

Reply:
yes. I have updated this.